### DISCUSSION TITLE
Hack night March 2021

### DISCUSSION THREAD

**Discussion by froi** [Code4PuertoRico/cfpr-website#2] [General]
Team discussion on our first hack night.

- Use discussions for general topics and ideas.
- Use Discussions for questions
- Use GitHub Issues for actionable items (work)

A discussion was created to brainstorm on what stack to use https://github.com/Code4PuertoRico/code4puertorico.github.io/discussions/2

**Comment by jicruz96** [comment]
Here are just a few thoughts:

* Bilingual site is a must for our global community.
* An About page that displays the org's mission and also showcases the current projects the brigade is working on
* A Get Involved area that holds the user hand as much as possible and helps them onboard (event calendar, slack channel, code of conduct, case studies from previous or ongoing projects, a contact page)
* A blog where C4PR makes announcements, press releases, shares content from friends (wasnt there a Puerto Rico tech podcast happening?)
* Perhaps a "Friends of C4PR" section somewhere that showcases partner projects or other orgs in the ecosystem.
* **A donate button** (take bitcoin lol)

**Comment by froi** [comment]
Ade mentions:

- UX is must
- Content to drive engagement
- Bigger players like Doctors without borders have cards that express their mission / efforts.
  - as a user clicking on these cards and getting relevant information on their efforts is engaging
- Thinks that Figma is an excellent tool
- Not over engineer navigation
- Tell a story
- Have a flow to the site
- Have a call to action
- Be explicit on our values

**Comment by froi** [comment]
Ebenezer mentions:

- Straight forward concept
- Be clear that we are just not code and not all of use are coders
- We are all in a common goal of helping where and how we can
- Learning resources
  - How can I learn to code
- Be clear in our messaging that you don't have to be in Puerto Rico to participate
- Be the anchor for the community
  - SEO

**Reply by ebetron** [reply]
Messaging: Code4PuertoRico is a multi-disciplinary effort (Coding, Design, Marketing, etc.)

**Comment by froi** [comment]
Layshi mentions:

- Case studies should be included on what we do
- How can I contribute
  - Volunteer matching
  - What efforts are currently available
  - Categories (Eg. domestic violence, government. etc)
- What skills are needed.
- Make everything inclusive
  - Not just accessibility but other communities
  - How do I fit
  - Tone of voice. How to communicate clearly
- As an example the Peace Corp.
  - They have stories.
  - Helps with recruitment
- Specify location for the efforts

**Reply by LaysCurbelo** [reply]
- Communications sections with a Media Kit that contains; About of organization, mission, vision, branding material (logo), press releases.

**Reply by LaysCurbelo** [reply]
An example of a section for recruiting people for the initiatives.  

![Screen Shot 2021-03-02 at 7 20 47 PM](https://user-images.githubusercontent.com/60011942/109728402-7bf19580-7b8c-11eb-980a-9f81127ae931.png)

**Reply by LaysCurbelo** [reply]
![Screen Shot 2021-03-02 at 7 25 56 PM](https://user-images.githubusercontent.com/60011942/109728717-25388b80-7b8d-11eb-8521-d9be147102bb.png)

This maybe is too advance at this moment but I think is important to show how people can participate with us, having an onboarding could be an awesome way to explain more "hard" topics like code of conduct, our values, and the way to proceed on Github.

**Comment by froi** [comment]
Dominique mentions

- Echos much of what was said
- As an outsider looking in he would like to see how to get involved
  - what we are about
- How to express that they are included
- A good flow to get people from the site to participation

**Comment by froi** [comment]
Ruben mentions

- We have the talent in PR
- Project discoverability is a big need
  - Finding a community
- Content is a big issue
- Highlighting the different personas that can help
  - People that are willing to write content
  - Marketing 
  - Translators
- Before having a call to action like join or help we should have the framework for people to answer what they want to do
- A clear map to where to get resources and efforts
- Have a clear place to find out what is happening
- Not needed access to know what's going on

**Reply by rubenvarela** [reply]
make content and assets available and easy to find

- Data
- datasets
- endpoints
- example code
- projects
- design assets
- design guidelines 
- styleguide

**Reply by rubenvarela** [reply]
Possible roles/personas we'd like to address on the site with more specific information and/or call to actions. 

- Developers
  - Projects
  - Datasets
  - Groups
- Designers
  - UI
  - UX
  - Styleguides
  - Assets
  - Contacts
  - Projects looking for help
  - Projects needing help
- Content
  - Copywriters
  - Review of content for issues (accessibility, inclusivity, terms, translations, etc.)
- Industry experts (to help document/answer things)
- Community Outreach/Efforts
- Marketing

**Comment by froi** [comment]
- How can we make adding content to the site. 
- Do we need a CMS? Could Gatsby be enough?
- How do we make it simple for people to contribute to the site via PRs?

**Reply by rubenvarela** [reply]
I think the main thing that will pose an issue is users that don't have a Github account. We'd need a form that would allow us to create the issue and manage it there afterwards. This would basically require a backend, somewhere.

Regarding stack, I personally would go for anything that compiles to static. Easy to manage, deploy and handoff in the future. Can be recreated by anyone.
The main thing that could pose an issue is search (usually the search on static sites is keyword based or maybe just includes stemming).

For backend, I think validating a captcha and posting if valid to a GitHub issue. Another option is honeypot fields. Both would need testing for a11y. 
We could look into some WYSIWYG on the form that does markdown. Or just having a template on a basic form. If plaintext, we'd have to be careful about the text formatting. 
If WYSIWYG, maybe something like, https://github.com/nhn/tui.editor (either we check it's a11y accessible or allowing fallback to just plaintext)

**Reply by kpl007** [reply]
https://jekyllrb.com/ might be an option as well.

**Reply by rubenvarela** [reply]
Definitely!

**Reply by dom-wr** [reply]
I'm a fan of either of those

**Comment by dom-wr** [comment]
Moving forward, we will need the slack link to contribute as well

**Reply by rubenvarela** [reply]
for now, https://code4pr.slack.com

**Reply by froi** [reply]
Specifically the `#webisite` channel.

**Reply by jamesxv7** [reply]
In other projects I have had situations that participants often share documents through Google Docs, maybe we can also explore adding this option to the flow.
